starter_code/canvas.js

```diff
@@ -1,32 +1,235 @@
+var hangman = new Hangman();
+var secretWord = hangman.getWord();
+var startBtn = document.getElementById("start-game-button");
 
 function HangmanCanvas(secretWord) {
-  this.ctx = document.getElementById('hangman').getContext('2d');
+  this.element = document.getElementById("hangman");
+  this.ctx = this.element.getContext("2d");
+  this.ctx.font = "30px sans-serif";
+  this.ctx.lineWidth = 6;
 }
 
-HangmanCanvas.prototype.createBoard = function () {
+HangmanCanvas.prototype.createBoard = function() {
+  this.cleanCanvas();
+  this.drawLines(secretWord.length);
+};
 
+HangmanCanvas.prototype.cleanCanvas = function() {
+  this.ctx.clearRect(0, 0, 1200, 800);
 };
 
-HangmanCanvas.prototype.drawLines = function () {
+HangmanCanvas.prototype.drawLines = function(numLines) {
+  this.ctx.beginPath();
 
-};
+  var positionX = 320;
+  var positionY = 780;
 
-HangmanCanvas.prototype.writeCorrectLetter = function (index) {
+  for (var i = 0; i < numLines; i++) {
+    positionX += 10;
+    this.ctx.moveTo(positionX, positionY);
 
+    positionX += 50;
+    this.ctx.lineTo(positionX, positionY);
+  }
+  this.ctx.stroke();
+  this.ctx.closePath();
 };
 
-HangmanCanvas.prototype.writeWrongLetter = function (letter, errorsLeft) {
+HangmanCanvas.prototype.writeCorrectLetter = function(index) {
+  var positionX = 340;
+  var positionY = 770;
 
+  this.ctx.fillText(secretWord[index], positionX + 60 * index, positionY);
 };
 
-HangmanCanvas.prototype.drawHangman = function (shape) {
+HangmanCanvas.prototype.writeWrongLetter = function(letter, errorsLeft) {
+  var positionX = 600;
+  var positionY = 120;
 
+  this.ctx.fillText(letter, positionX + (10 - errorsLeft) * 35, positionY);
 };
 
-HangmanCanvas.prototype.gameOver = function () {
-
+HangmanCanvas.prototype.drawHangman = function(shape) {
+  switch (shape) {
+    case 1:
+      var newFun = printBase.bind(this);
+      newFun();
+      break;
```

Comment:
Os podeis evitar el bind(this) y la llamada a la función a continuación, si pasais el ctx como parámetro
`printBase(this.ctx);`

Reply:
Hay algun problema o desventaja en usar bind()?